#  <p style="text-align: center;">Практическое задание к Семинару 1.<br><br>Урок 1. Расчет вероятности случайных событий.</p>


In [88]:
from math import factorial

**Формула вероятности:**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>
$\displaystyle P(A)=\frac{m}{n}$ (1)<br>, где $m$-число благоприятных исходов, $n$-общее число всех исходов.</b></font></div>


**Формула подсчета количества сочетаний по $k$ элементов из множества $n$:**  
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$\displaystyle C^k_n=\frac{n!}{k!(n-k)!}$ (2)</b></font></div>

## ЗАДАНИЕ 1. 
**Из колоды в 52 карты извлекаются случайным образом 4 карты.**<br>
```
a) Найти вероятность того, что все карты – крести.
б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.
```

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

В данной задаче ПОРЯДОК неважен и участвуют НЕ ВСЕ ЭЛЕМЕНТЫ, то задачу будем решать формулой **СОЧЕТАНИЙ**.

**a) Найти вероятность того, что все карты – крести.**
1) Находим все сочетания из 4 карт возможные в колоде из 52 карт.
2) Находим все сочетания 4 карт одной масти (учитываем, что в колоде 4 масти, тогда $\frac{52}{4}=13$).
3) Находим вероятность вытащить 4 карты одной масти.

Исходя из формулы 2, найдем:<br>1) $n$ - общее число всех исходов или все сочетания из 4 карт возможные в колоде из 52 карт: $\displaystyle n=C^4_{52}$;<br> 
2) $m$ - число благоприятных исходов или все сочетания из 4 карт одной масти (учитываем, что в колоде 4 масти, тогда $\frac{52}{4}=13$ ): $\displaystyle m=C^4_{13}$

In [89]:
def sochetaniy(n, m): #функция расчета количества сочетаний
    return int(factorial(n) / (factorial(m) * factorial(n - m)))

In [90]:
# n - все возможные исходы
n_1=sochetaniy(52, 4)
print(f'n = {n}')

n = 270725


In [91]:
# m - все благоприятные исходы
m_1=sochetaniy(13, 4)
print(f'm = {m}')

m = 715


In [92]:
# P - вероятность
P_1=(m_1/n_1)
print(f'P = {P_1}')

P = 0.0026410564225690276


### 2 способ. Решение задачи через формулу вероятности.
Рассчитаем вероятность вытащить крестовую карту, для каждой попытки:<br>
$\displaystyle P_{1-карта}=\frac{13}{52}$;
$\displaystyle P_{2-карта}=\frac{12}{51}$;
$\displaystyle P_{3-карта}=\frac{11}{50}$; 
$\displaystyle P_{4-карта}=\frac{10}{49}$.|  
$\displaystyle P(4 крестовые)=\frac{13}{52}\cdot\frac{12}{51}\cdot\frac{11}{50}\cdot\frac{10}{49}$   

In [93]:
P_1_2=13/52*12/51*11/50*10/49
print(f'P(4крестовые) = {round(P_1_2,4)}')

P(4крестовые) = 0.0026


**б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.**
1) Находим все сочетания из 4 карт возможные в колоде из 52 карт.  => $\displaystyle n=C^4_{52}$
2) Находим все сочетания 4 карт, в которых нет ни одного туза (в колоде 4 туза разной масти, тогда 52-4=48) => $\displaystyle n=C^4_{48}$
3) Находим вероятность вытащить хотя бы один туз. Учтем,что сумма вероятностей противоположных событий равна 1:<br>  $\displaystyle P(A)+P(\bar{A})=1$, т.е. количество способов извлечь 4 карты из колоды **без** тузов. Тогда $\displaystyle P(A)=1-P(\bar{A})$

Исходя из формулы 2, найдем:<br>1) $n$ - общее число всех исходов или все сочетания из 4 карт возможные в колоде из 52 карт: $\displaystyle n=C^4_{52}$;<br> 
2) $m$ - число благоприятных исходов или все все сочетания карт без тузов: $\displaystyle m=C^4_{48}$

In [94]:
# n_1_b - все возможные сочетания
n_1_b=sochetaniy(52, 4)
print(f'n = {n_1_b}')

n = 270725


In [95]:
# m_1_b - все сочетания карт без тузов
m_1_b=sochetaniy(48, 4)
print(f'm = {m_1_b}')

m = 194580


In [96]:
# P - вероятность
P_tyz=1-(m_1_b/n_1_b)
P_tyz

0.2812632745405854

### 2 способ.
Выше определили, что общее число исходов $\displaystyle n=C^4_{52} = 270725_{исходов}$  
Вычислим $\displaystyle m$, используя формулу вероятности => $\displaystyle P(A)=\frac{m}{n}$ для каждого возможного варианта и извлечения:  
   **1-й вариант** - выташили из 4-х карт 1 туз:  
   а) находим сочетания достать 1 туз из 4-х $C^1_4$,<br>б) находим сочетания достать три карты (не туза) $С^3_{48}$,<br>в) находим общее количество сочетаний, где 1 туз и 3 не туза $C^1_{4}\cdotС^3_{48}$<br><br>
   **2-й вариант** - выташили из 4-х карт 2 туза:  
   а) находим сочетания достать 2 туза из 4-х $C^2_4$,<br>б) находим сочетания достать две карты (не туза) $С^2_{48}$,<br>в) находим общее количество сочетаний 2 туза и 2 не туза $C^2_{4}\cdotС^2_{48}$<br><br>
   **3-й вариант** - выташили из 4-х карт 3 туза:  
   а) находим сочетания достать 3 туза из 4-х $C^3_4$,<br>б) находим сочетания достать одну карту (не туза) $С^1_{48}$,<br>в) находим общее количество сочетаний 3 туза и 1 не туз $C^3_{4}\cdotС^1_{48}$<br><br>
   **4-й вариант** - выташили из 4-х карт 4 туза:  
   а) находим сочетания достать 4 туза из 4-х $C^4_4=1$  
   
 $m$ - число благоприятных исходов = сумме сочетаний всех 4 вариантов.  

In [97]:
m_1_b_1=sum([sochetaniy(4,1)*sochetaniy(48,3),sochetaniy(4,2)*sochetaniy(48,2),sochetaniy(4,3)*sochetaniy(48,1),1])
print(f'm = {m_1_b_1}')

m = 76145


In [98]:
# P - вероятность
P_tyz_1=m_1_b_1/n_1_b
P_tyz_1

0.2812632745405855

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>  а) Вероятность вытащить все карты масти крести равна $P\approx 0.0026 \approx 0.26\%$.<br>б) Вероятность, что среди 4-х карт окажется хотя бы один туз равна $P\approx 0.2813 \approx 28.13\%$.</font></div>

---

## ЗАДАНИЕ 2. 
**На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

В данной задаче ПОРЯДОК неважен и участвуют НЕ ВСЕ ЭЛЕМЕНТЫ, то задачу будем решать формулой **СОЧЕТАНИЙ**.
1) Найдем **n** - общее число всех исходов.
2) Найдем **m** -  число благоприятных исходов.
3) Найдем **P** - вероятность, что человек, не знающий код, откроет дверь с первой попытки.

Исходя из формулы 2, найдем:<br>1) $n$ - общее число всех исходов (цифр 10 и 3 кнопки надо нажать одновременно): $\displaystyle n=C^3_{10}$;<br> 
2) $m$ - число благоприятных исходов или все все сочетания карт без тузов: $\displaystyle m=1$

In [99]:
# n - общее число всех исходов (цифр 10 и 3 кнопки надо нажать одновременно)
n_2=sochetaniy(10, 3)
print(f'n = {n_2}')

n = 120


In [100]:
# m - все благоприятные исходы, т.к. нам надо найти вероятность того, что человек откроет дверь с ПЕРВОЙ попытки, то у нас есть только ОДИН благоприятный исход.
m_2 = 1
print(f'm = {m_2}')

m = 1


In [101]:
# P - вероятность
P_2=m_2/n_2
P_2

0.008333333333333333

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>Вероятность открыть дверь с <b>ПЕРВОЙ</b> попытки одновременным нажатием <b>ТРЁХ</b> кнопок (цифр): $\displaystyle P\approx0.0083\approx0.83\%$</font></div>

---

## ЗАДАНИЕ 3. 
**В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

В данной задаче ПОРЯДОК неважен и участвуют НЕ ВСЕ ЭЛЕМЕНТЫ, то задачу будем решать формулой **СОЧЕТАНИЙ**.
1) Найдем **n** - общее число всех исходов.
2) Найдем **m** - число благоприятных исходов.
3) Найдем **P** - вероятность, что извлеченные детали окрашены.

Исходя из формулы 2, найдем:<br>1) $n$ - общее число всех исходов (всего количество деталей 15 штук, извлекаются 3 детали): $\displaystyle n=C^3_{15}$;<br> 
2) $m$ - все благоприятные исходы (всего окрашенных деталей 9 и надо извлечь 3 детали): $\displaystyle m=C^3_{9}$

In [102]:
# n - общее число всех исходов (всего количество деталей 15 штук, извлекаются 3 детали)
n_3=sochetaniy(15, 3)
print(f'n = {n_3}')

n = 455


In [103]:
# m - все благоприятные исходы (всего окрашенных деталей 9 и надо извлечь 3 детали)
m_3=sochetaniy(9, 3)
print(f'm = {m_3}')

m = 84


In [104]:
# P - вероятность
P_3=m_3/n_3
P_3

0.18461538461538463

### 2 способ. Решение задачи через формулу вероятности.
Рассчитаем вероятность вытащить крестовую карту, для каждой попытки:<br>
$\displaystyle P_{1-окраш}=\frac{9}{15}$;
$\displaystyle P_{2-окраш}=\frac{8}{14}$;
$\displaystyle P_{3-окраш}=\frac{7}{13}$.|  
$\displaystyle P(4 окраш)=\frac{9}{15}\cdot\frac{8}{14}\cdot\frac{7}{13}$  

Или $P(\text{3 из 3-х окрашены})=\frac{9}{15}\cdot\frac{8}{14}\cdot\frac{7}{13}$

In [105]:
P_3_1=9/15*8/14*7/13
P_3_1

0.1846153846153846

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>Вероятность вытащить с <b>ПЕРВОЙ</b> попытки 3 окрашенные детали: $\displaystyle P\approx0.1846\approx18.46\%$</font></div>

---

## ЗАДАНИЕ 4. 
**В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

В данной задаче ПОРЯДОК неважен и участвуют НЕ ВСЕ ЭЛЕМЕНТЫ, то задачу будем решать формулой **СОЧЕТАНИЙ**.
1) Найдем **n** - общее число всех исходов.
2) Найдем **m** - число благоприятных исходов.
3) Найдем **P** - вероятность, что извлеченные детали окрашены.

Исходя из формулы 2, найдем:<br>1) $n$ - общее число всех исходов (всего билетов 100 штук, извлекаются 2 билета): $\displaystyle n=C^2_{100}$;<br> 
2) $m$ - все благоприятные исходы, т.к. нам надо найти вероятность того, что приобретаем билеты с первого раза: $\displaystyle m=1$

In [106]:
# n - общее число всех исходов (всего билетов 100 штук, извлекаются 2 билета)
n_4=sochetaniy(100, 2)
print(f'n = {n_4}')

n = 4950


In [107]:
# m - все благоприятные исходы, т.к. нам надо найти вероятность того, что приобретаем билеты с ОДИН раз.
m_4 = 1
print(f'm = {m_4}')

m = 1


In [108]:
# P - вероятность
P_4=m_4/n_4
P_4

0.00020202020202020202

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>Вероятность того, что 2 приобретенных билета окажутся выигрышными: $\displaystyle P\approx0.0002\approx0.02\%$</font></div>